sktime/dists_kernels/compose_tab_to_panel.py

```diff
@@ -58,18 +67,21 @@ def _transform(self, X, X2=None):
         n = len(X)
         m = len(X2)
 
-        symmetric = self.symmetric
+        X_equals_X2 = self.X_equals_X2
 
         aggfunc = self.aggfunc
-        if aggfunc is None:
-            aggfunc = np.mean
+        aggfunc_symm = self.aggfunc_symm
+        transformer_symm = self.transformer._all_tags()["symmetric"]
+
+        # whether we know that resulting matrix must be symmetric
+        all_symm = aggfunc_symm and transformer_symm and X_equals_X2
```

Comment:
Minor:
The logic here is non-trivial (but I think it is correct). I don't know how, but might be worth documenting this logic somewhere.

Reply:
I added a comment for this